redesign.user.js

```diff
@@ -2579,8 +2579,8 @@ const SETTINGS = [
     new BooleanSetting('general.googlyEyes', true),
     new BooleanSetting('general.semesterzeiten', false),
     new SelectSetting('general.prideLogo', 'default', [
-        'off',
-        'default',
+        'false',
+        'true',
```

Comment:
Das funktioniert nicht so wie vermutlich beabsichtigt, weil `false !== 'false'`.

Reply:
Wenn ich den Wert `false` oder `true` nenne, wird der drop-down selector `undefined`

Reply:
Deswegen würde ich persönlich die Einstellung umbenennen und beim den Wert der alten Einstellung in die neue umwandeln, wenn die Alte existiert. 
Es ist auch möglich zu prüfen, ob in der Einstellung ein boolean ist und entsprechend dann den richtigen String auszuwählen.

Reply:
Uff, ich glaube da reichen meine begrenzten JS Kenntnisse nicht für aus.
Ich denke, ich könnte in den nächsten Tagen weiter daran herumbasteln, aber ich verstehe auch, wenn euch das zu viel Arbeit macht, das alles zu prüfen. 
Ich bin generell nicht sehr erfahren was git angeht, also ist es auch kein Problem, wenn ihr die Request schließt

Reply:
ich wollte dich nicht demotivieren. Wenn du magst forke ich deinen fork und baue daran weiter, denn die Idee finde ich super!

Reply:
Wenn du das möchtest, ist das kein Problem, ich dachte von Anfang an, das Ganze wäre etwas einfacher (mit Gradienten). An sich bin ich natürlich motiviert daran weiterzuarbeiten, aber für viele der backend Sachen fehlt mir einfach die Erfahrung das korrekt umzusetzen

Reply:
> Deswegen würde ich persönlich die Einstellung umbenennen und beim den Wert der alten Einstellung in die neue umwandeln, wenn die Alte existiert. Es ist auch möglich zu prüfen, ob in der Einstellung ein boolean ist und entsprechend dann den richtigen String auszuwählen.

Wäre das sowas wie:
`if (getSetting('general.prideLogo') === false){
    GM_setValue(getSettingKey('general.prideLogo'), 'off');
}`
und entsprechend äquivalent für `true`?

Reply:
Man müsste auch noch überprüfen, ob der Wert der neuen Einstellung schon gesetzt wurde.
Anderer Vorschlag: Default-Value der neuen Einstellung abhängig von dem was im Speicher steht setzen. Da können wir nicht mit getSetting arbeiten, aber mit GM_getValue etc. sollte das möglich sein.

Reply:
Würde
`/**
* Handles legacy transition for the prideLogo settings
* @returns {string} default setting
*/
const updatePrideSetting = () => {
    const currentValue = GM_getValue(getSettingKey('general.prideLogo'), 'default');
    if(currentValue === true){
        return 'default'
    }else if(currentValue === false){
        return 'off'
    }else{
        return 'default'
    }
}`
als helper Funktion funktionieren?
Und dann ` new SelectSetting('general.prideLogo', updatePrideSetting(), [ ... ]`

Oder ist das ein komplett falscher Weg das zu Implementieren?

Reply:
Ich würde's noch etwas einfacher halten und ohne helperFunction, sollte eigentlich funktionieren, wenn ich nicht ganz falsch denke:
```js
new SelectSetting('general.prideLogo', GM_getValue(getSettingKey('general.prideLogo'), true).toString(), [...]),
```
Das sollte ja theoretisch alle Fälle abdecken:
* Ist die (neue) Einstellung schon mal gesetzt worden, wird das ganz normal ausgelesen
* Ist die alte Einstellung nie gespeichert worden (z. B. Upgrade mit Version-skip), dann nimmt er `'true'` als default value
* Ist die alte Einstellung mal gesetzt worden, dann stringified er diese zu `'true'` bzw. `'false'`

Hab ich da was übersehen?

Eine HelperFunction ist natürlich grundsätzlich auch möglich, aber gefällt mir nicht ganz so gut. Eine Utility zum nachträglichen Ändern von Einstellungen (Wo man dann Migrations-Funktionen angeben kann oder so) würde mir langfristig besser gefallen. (Merke ich mir für #168 bzw. den `v2`-branch).

Reply:
Das ist natürlich deutlich eleganter. 
Hab ich umgesetzt, scheint so weit zu funktionieren.
Danke :)